docs/_guide/templates.md

```diff
@@ -569,6 +581,162 @@ html`${until(content, html`<span>Loading...</span>`)}`
+}
+```
+
+LitElement supplies a set of decorators that provide a shorthand way of defining getters like this.
+
+More information:
+
+*   [Element.querySelector()](https://developer.mozilla.org/en-US/docs/Web/API/Element/querySelector) on MDN.
+*   [Element.querySelectorAll()](https://developer.mozilla.org/en-US/docs/Web/API/Element/querySelectorAll) on MDN.
+
+### @query, @queryAll, and @queryAsync decorators
+
+The `@query`, `@queryAll`, and `@queryAsync` decorators all provide a convenient way to access nodes in the component's shadow root. 
+
+<div class="alert alert-info">
+
+**Using decorators.** Decorators are a proposed JavaScript feature, so you’ll need to use a compiler like Babel or the TypeScript compiler to use decorators. See [Using decorators](decorators) for details.
+
+</div>
+
+The `@query` decorator modifies a class property, turning it into a getter that returns a node from the render root.
+
+```js
+class MyElement {
+  @query('#first')
+  first;
+
+  render() {
+    return html`
+      <div id="first"></div>
+      <div id="second"></div>
+    `;
+  }
+}
+```
+
+This decorator is equivalent to:
+
+```js
+get first() {
+  return this.renderRoot.querySelector('#first');
+}
+```
+
+<div class="alert alert-info">
+
+**shadowRoot and renderRoot**. The [`renderRoot`](/api/classes/_lit_element_.litelement.html#renderroot) property identifies the container that the template is rendered into. By default, this is the component's `shadowRoot`. The decorators use `renderRoot`, so they should work correctly even if you override `createRenderRoot` as described in [Specify the render root](#renderroot)
+
+</div>
+
+The `@queryAll` decorator is identical to `query` except that it returns all matching nodes, instead of a single node. It's the equivalent of calling `querySelectorAll`.
+
+
+```js
+class MyElement {
+  @queryAll('div')
+  divs;
+
+  render() {
+    return html`
+      <div id="first"></div>
+      <div id="second"></div>
+    `;
+  }
+}
+```
+
+Here, `divs` would return both `<div>` elements in the template. For TypeScript, the typing of a `@queryAll` property is `NodeListOf<HTMLElement>`. If you know exactly what kind of nodes you'll retrieve, the typing can be more specific:
+
+```
+@queryAll('button')
+buttons!: NodeListOf<HTMLButtonElement>
+```
+
+The exclamation point (`!`) after `buttons` is TypeScript's [non-null assertion operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator). It tells the compiler to treat `buttons` as always being defined, never `null` or `undefined`.
+
+Finally, `@queryAsync` works like `@query`, except that instead of returning a node directly, it returns a `Promise` that resolves to that node. External code can use this instead of waiting for the `updateComplete` promise. 
+
+This is useful, for example, if the node returned by `@queryAsync` can change as a result of another property change. 
```

Comment:
Maybe @sorvell can add some insight on documenting `@queryAsync` externally. It's most for internal tool usage, and I'm a little worried that it seems like we're encouraging exposing shadow DOM contents publicly here. I wonder how we can describe this well, or if we should just ignore `@queryAsync` in the main docs.

Reply:
By "internal tool usage" do you mean for testing? I'm not actually too clear on the use cases for this decorator. 

Help me @sorvell you're my only hope!

Reply:
Sorry, that comment was unclear.

I was imagining the use case where a dev decides to expose a public property via `@query('.foo') fooElement`. In order for that to work correctly at all times, the user would have to `await element.updateComplete`:

```
await element.updateComplete;
element.fooElement.textContent = 'hi';
```
Instead the dev may choose to expose the property with `@queryAsync('.foo') fooElement;` so the user code can be more compact:

```
(await element.fooElement).textContent = 'hi';
```

Reply:
OK, understood. I think the concern @justinfagnani had was that these properties that expose shadow DOM children should probably be internal so we don't leak implementation details.

That said, a property getter provides more controlled access than having the user code call `element.shadowRoot.querySelector()` themselves. So...

Reply:
For now, just leaving this in with the word "external" removed.